### PR TITLE
[JMENano] Add customize functions to recompute Puppi weights and PuppiMET, add new autoNano option

### PR DESCRIPTION
**Pull request by nurfikri89**
#### PR description:

This PR adds three wrapper functions which 

- set the PuppiProducer instances (`packedpuppi` & `packedpuppiNoLep`) to recompute puppi weights.
- setup the recalculation of PuppiMET.
- wraps over the two functions above.

The three functions can be set up in cmsRun config files  using `--customise` or `-customize_commands` arguments of cmsDriver.py. This would be useful if a (official or private) JMENano production needs updated Puppi weights. An `autoNano` option called `jmeNANOrePuppi` is added which sets up JMENano production with recomputation of Puppi weights and Puppi MET. Added two new test workflows (`2500.328,2500.406`) using the new `autoNano` option.

#### PR validation:

passes the usual runTheMatrix test: `runTheMatrix.py -l limited -i all --ibeos`
passes the new JMENano wokflows runTheMatrix: `runTheMatrix.py -i all --ibeos -l 2500.328,2500.406`

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported to 14_0_X

### CODE REVIEW

**Comment by vlimant** [comment]
enable nano

**Comment by vlimant** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_nano.py [hunk 1]
```diff
@@ -224,6 +227,8 @@ def subnext(self):
 steps['jmeNANO_mc13.2']=merge([{'-s' : 'NANO:@JME ', '-n' : '1000'},
                                     steps['NANO_mc13.2']])
 
+steps['jmeNANO_rePuppi_mc13.2']=merge([steps['jmeNANO_mc13.2']])
```
Comment (vlimant) [inline Configuration/PyReleaseValidation/python/relval_nano.py:230]: please use the `@JMErePuppi`

File: Configuration/PyReleaseValidation/python/relval_nano.py [hunk 2]
```diff
@@ -192,13 +192,16 @@ def subnext(self):
...
-steps['jmeNano_data13.0'] = merge([{'-s':'NANO:@JME', '-n' : '1000'},
+steps['jmeNANO_data13.0'] = merge([{'-s':'NANO:@JME', '-n' : '1000'},
                                  steps['NANO_data13.0']])
 
 steps['lepTrackInfoNANO_data13.0']=merge([{'-s' : 'NANO:@LepTrackInfo,DQM:@nanoAODDQM',
                                           '-n' : '1000'},
                                          steps['NANO_data13.0']])
 
+steps['jmeNANO_rePuppi_data13.0']=merge([steps['jmeNANO_data13.0']])
+steps['jmeNANO_rePuppi_data13.0']['--customise'] += ',"PhysicsTools/NanoAOD/custom_jme_cff.RecomputePuppiWeightsAndMET"'
```
Comment (vlimant) [inline Configuration/PyReleaseValidation/python/relval_nano.py:203]: please use the `@JMErePuppi`

File: Configuration/PyReleaseValidation/python/relval_nano.py (hunk 1 above)
Comment (nurfikri89) [reply Configuration/PyReleaseValidation/python/relval_nano.py:230]: Ah I missed this. Will fix it now.

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
+1

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by srimanob** [comment]
+Upgrade

**Comment by antoniovilela** [comment]
+1

**Pull request merged**